### ISSUE TITLE
Implement deep learning CNN for plate detection

### ISSUE THREAD

**Issue by matthill** [openalpr/openalpr#99]
Deep learning convolutional neural networks seem like the next generation for object detection.  The accuracy is significantly better, and it is not tied to particular aspect ratio.  This would work very well for places like India where the license plate formats vary significantly, and Europe and other countries that have 1-line and 2-line plates.  It also handles occlusion/rotation better.  The disadvantage is that it requires massive amounts of data and training time.

https://www.youtube.com/watch?v=S75EdAcXHKk
http://arxiv.org/pdf/1502.02766v2.pdf
http://caffe.berkeleyvision.org/

**Comment by cysin** [comment]
Looking forward to this, any plans?

**Comment by matthill** [comment]
I am too :)  I'm not sure if Caffe is mature enough for this use-case though.  I'm going to wait and see.  The Cascade detectors work pretty good right now.  This has a lot of potential though.

**Comment by Irtza** [comment]
Doing this in TensorFlow for my final year project. How can I contribute my work here? 
Anyone else looking forward to collaborate other outside this project. I'd like to hear from you

**Comment by mrgloom** [comment]
I can test some deep learning algorithms for object detection have you any data with markup ?

**Comment by matthill** [comment]
What kind of data are you looking for?  There's some pos/neg samples in this repo used for cascade training: https://github.com/openalpr/train-detector

**Comment by KowszewiczGrzegorz** [comment]
Any development here?

**Comment by OmarMAmin** [comment]
@Irtza  have you came up with any thing from your project ?

**Comment by chaitan94** [comment]
There seems to be an experimental implementation of this here: https://github.com/matthewearl/deep-anpr with more details in his [blog](https://matthewearl.github.io/2016/05/06/cnn-anpr/). It uses [TensorFlow](https://tensorflow.org/). This not only does plate detection but also recognizes the information on the plate as well. Though I am not sure how it compares to openalpr in terms of accuracy.

While it is still in alpha, even Tesseract in its new version (4.0) has moved on to make LSTM as their default engine going forward:
https://github.com/tesseract-ocr/tesseract/wiki/4.0-with-LSTM
https://github.com/tesseract-ocr/tesseract/wiki/NeuralNetsInTesseract4.00

If openalpr did move to a neural network based approach, it might need to add a dependency on a framework like TensorFlow. Would it be justified for just a single step (plate detection)?

I am not very experienced with this field yet, but am interested enough to contribute if possible. I would like to know the thoughts of someone who has worked with similar projects, or from the openalpr maintainers on this issue about what would be an appropriate approach.

**Comment by sumsuddin** [comment]
How about [this](https://arxiv.org/pdf/1709.08828.pdf) one? Looks interesting.
